Microsoft.Toolkit.Uwp.Services/Services/MicrosoftGraph/Drive/OneDriveItemExtension.cs

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Graph;
+using Newtonsoft.Json;
+using Windows.Storage.Streams;
+using static Microsoft.Toolkit.Uwp.Services.MicrosoftGraph.MicrosoftGraphEnums;
+
+namespace Microsoft.Toolkit.Uwp.Services.MicrosoftGraph
+{
+    /// <summary>
+    ///  Class of Microsoft Graph API Extension
+    /// </summary>
+    public static class OneDriveItemExtension
+    {
+        /// <summary>
+        /// Is the DriveItem is type of file
+        /// </summary>
+        /// <param name="onedriveItem">An instance of OneDriveStorageItem</param>
+        /// <returns>True if it's a file otherwise false</returns>
+        public static bool IsFile(this OneDriveStorageItem onedriveItem)
+        {
+            return onedriveItem.OneDriveItem.File != null ? true : false;
+        }
+
+        /// <summary>
+        /// Is the DriveItem is type of folder
+        /// </summary>
+        /// <param name="onedriveItem">An instance of OneDriveStorageItem</param>
+        /// <returns>True if it's a folder otherwise false</returns>
+        public static bool IsFolder(this OneDriveStorageItem onedriveItem)
+        {
+            return onedriveItem.OneDriveItem.Folder != null ? true : false;
+        }
+
+        /// <summary>
+        /// Creates a copy of the item in the specified folder and renames the item according to the desired name.
+        /// </summary>
+        /// <param name="builder">Http request builder</param>
+        /// <param name="provider">Http provider to execute the request</param>
+        /// <param name="destinationFolder">The folder where the item will be created</param>
+        /// <param name="desiredNewName">The new name for the copy of the item created in the destination Folder</param>
+        /// <returns>return success or failure</returns>
+        public static Task<bool> CopyAsync(this IDriveItemRequestBuilder builder, GraphServiceClient provider, OneDriveStorageFolder destinationFolder, string desiredNewName)
+        {
+            var requestUrl = builder.RequestUrl + "/copy";
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, requestUrl);
```

Comment:
I'm not sure Eric will be able to change it. Do you want to push the fix on his branch?